pkg/ddl/partition.go

```diff
@@ -1399,21 +1417,22 @@ func buildRangePartitionDefinitions(ctx sessionctx.Context, defs []*ast.Partitio
 				return nil, exprChecker.err
 			}
 			// If multi-column use new evaluated+normalized output, instead of just formatted expression
-			if len(partValStrings) > i && len(colTypes) > 1 {
+			if len(partValStrings) > i && len(tbInfo.Partition.Columns) > 0 {
```

Comment:
Not sure whether it's correct to replace `len(colTypes) > 1` with `len(tbInfo.Partition.Columns) > 0`. Is the number `1` a special number or just a typo?

Is it also correct to only set `len(partValStrings) > i` without other conditions here? @mjonss PTAL

Reply:
Above there is a check for `if len(colTypes) != len(tbInfo.Partition.Columns)` so they should be the same here.
I think there was a reason for keeping the output for single column RANGE COLUMNS partitioning (IIRC the evaluation+normalization of the values was done when adding multi-column RANGE COLUMNS, so I just wanted to avoid changing the single column at the same time).

So if we also want the evaluated+normalized value string here also for single column RANGE COLUMNS, then it is OK to also remove the `&& len(tbInfo.Partition.Columns)` part.
I.e. OK to change now so also single column RANGE COLUMNS uses the better formatting.

Reply:
OK. I'll remove this condition. Then only partition with simple expression (and without column) will use `expr.Format`, all other situations will use the evaluated value.